pkg/generate/code/compare_helpers.go

```diff
@@ -0,0 +1,460 @@
+package code
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	awssdkmodel "github.com/aws/aws-sdk-go/private/model/api"
+
+	ackgenconfig "github.com/aws-controllers-k8s/code-generator/pkg/generate/config"
+	"github.com/aws-controllers-k8s/code-generator/pkg/model"
+	"github.com/aws-controllers-k8s/code-generator/pkg/names"
+)
+
+// CompareResourceHelpers returns Go code that define the helper functions used by
+// newResourceDelta function. The helper functions return true if two given parameters
+// are equal,false otherwise.
+func CompareResourceHelpers(
+	cfg *ackgenconfig.Config,
+	r *model.CRD,
+	// Number of levels of indentation to use
+	indentLevel int,
+) string {
+	out := "\n"
+
+	specFieldNames := []string{}
+	for fieldName := range r.SpecFields {
+		specField := r.SpecFields[fieldName]
+		if expectHelperFunction(specField.ShapeRef.Shape) {
+		}
+		specFieldNames = append(specFieldNames, fieldName)
+	}
+	sort.Strings(specFieldNames)
+
+	for _, fieldName := range specFieldNames {
+		specField := r.SpecFields[fieldName]
+		memberShapeRef := specField.ShapeRef
+		memberShape := memberShapeRef.Shape
+		shapes := collectSubShapes(memberShape)
```

Comment:
Thinking that it might be better to collect the shapes that needs a helper method in `CompareResource` (during the `newResourceDelta` generation)

Reply:
I really do like that they're completely decoupled hmm... Would it be too crude to pass around a `[string]shape` map? I fear that using a partial `CRD` could cause issues if later on we accidentally try to use the other attributes.

Reply:
I was thinking about appending helper functions Go code to an array while generating the `newResourceDelta`. But i totally agree it will be very crude - especially if we conciser nested helper functions... Let's keep it decoupled :)